assets/js/components/DateRangeSelector.js

```diff
@@ -19,40 +19,92 @@
 /**
  * WordPress dependencies
  */
-import { useCallback } from '@wordpress/element';
+import { useCallback, useEffect, useRef, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
-import { Option, Select } from '../material-components';
+import DateRangeIcon from '../../svg/date-range.svg';
+import Menu from './menu';
 import { getAvailableDateRanges } from '../util/date-range';
 import { STORE_NAME as CORE_USER } from '../googlesitekit/datastore/user/constants';
+import Button from './button';
+import { map } from 'lodash';
 const { useSelect, useDispatch } = Data;
 
 function DateRangeSelector() {
 	const ranges = Object.values( getAvailableDateRanges() );
 	const dateRange = useSelect( ( select ) => select( CORE_USER ).getDateRange() );
 	const { setDateRange } = useDispatch( CORE_USER );
-	const onChange = useCallback( ( index, item ) => {
-		setDateRange( item.dataset.value );
-	}, [ ranges ] );
+
+	const [ menuOpen, toggleMenu ] = useState( false );
+	const menuButtonRef = useRef();
+	const menuRef = useRef();
+
+	useEffect( () => {
+		const handleMenuClose = ( e ) => {
+			// Close the menu if the user presses the Escape key
+			// or if they click outside of the menu.
+			if (
+				( ( 'keyup' === e.type && 27 === e.keyCode ) || 'mouseup' === e.type ) &&
+				! menuButtonRef.current.buttonRef.current.contains( e.target ) &&
+				! menuRef.current.menuRef.current.contains( e.target )
+			) {
+				toggleMenu( false );
+			}
+		};
+
+		global.addEventListener( 'mouseup', handleMenuClose );
+		global.addEventListener( 'keyup', handleMenuClose );
+
+		return () => {
+			global.removeEventListener( 'mouseup', handleMenuClose );
+			global.removeEventListener( 'keyup', handleMenuClose );
+		};
+	}, [] );
+
+	const handleMenu = useCallback( () => {
+		toggleMenu( ! menuOpen );
+	}, [ menuOpen ] );
+
+	const handleMenuItemSelect = useCallback( ( index, e ) => {
+		if (
+			( 'keydown' === e.type && ( 13 === e.keyCode || 32 === e.keyCode ) ) || // Enter or Space is pressed.
+			'click' === e.type // Mouse is clicked
+		) {
+			setDateRange( ranges[ index ].slug );
+			toggleMenu( false );
+		}
+	}, [ handleMenu ] );
+
+	const currentDateRangeLabel = ranges.reduce( ( acc, range ) => {
+		return range.slug === dateRange ? range : acc;
+	}, {} ).label;
+
+	const menuItems = map( ranges, 'label' );
```

Comment:
Instead of using `lodash.map`, let's use `Array.map` instead.

Historically there was a lot of lodash use in the app but moving forward we're preferring ES6 built-ins when possible. Lodash has some fancy helpers that are worth using, but for stuff like `map` I think `ranges.map( (range) => range.label )` is more immediately intuitive and doesn't require knowing the Lodash API—or at least as much of it 🙂

Reply:
Ok, that makes sense. I've switched to `Array.map`